### ISSUE TITLE
Is YOLO model using the validation set for training purpose?

### ISSUE THREAD

**Issue by pretiand** [pjreddie/darknet#2521]
For what concerns training, validation and test dataset I think there is a little bit of confusion in literature terminology.
I'm using yolov3 to train a model on custom 9 classes. 
I don't understand if the valid set provide in .data file is used in someway during the training process (i.e. tune hyperparameters, etc..) or it is used only to calculate some metrics (so as a sort of "test" set of UNSEEN data).

Thank you all!

**Comment by Kjelldor** [comment]
How I understand it Yolo will use the validation set for it's accuracy (mAP) calculations. It will not be used as training data, since most people won't be trying to detect something on exactly the same image as their training data.

By calculating the accuracy on images it has not trained with, it provides the user with a better estimation of real life performance.
